File: 백준/Silver/2559. 수열/수열.c
#include <stdio.h>
int main()
{
    int n,m;
    int i;
    int sum=0,max;
    int arr[100000];
    scanf("%d %d",&n,&m);
    for(i=0;i<n;i++)
        scanf("%d",&arr[i]);
    for(i=0;i<m;i++)
        sum+=arr[i];
    max=sum;
    for(i=1;i<=n-m;i++)
    {
        sum+=-arr[i-1]+arr[i+m-1];
        if(max<sum)
            max=sum;
    }
    printf("%d",max);
}

File: 백준/Silver/2491. 수열/수열.py
n = int(input())
arr = list(map(int,input().split()))

len1,len2 = 1,1
answer = 1
for i in range(n-1):
    if arr[i]<=arr[i+1]:
        len1 += 1
    else:
        len1 = 1
    if arr[i]>=arr[i+1]:
        len2 += 1
    else:
        len2 = 1
    answer = max(answer,len1,len2)

print(answer)